x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/WatcherLifeCycleService.java

```diff
@@ -203,7 +204,7 @@ private boolean clearAllocationIds() {
         return previousShardRoutings.get();
     }
 
-    public WatcherState getState() {
-        return state.get();
+    public Supplier<WatcherState> getState(){
```

Comment:
Need a live view of the state, but I didn't want to expose the actual reference to avoid the ability to set the state.